provider/bikeshed/src/units/essentials.rs

```diff
@@ -0,0 +1,140 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::collections::HashSet;
+
+use crate::cldr_serde::{self};
+use crate::DatagenProvider;
+
+use icu::experimental::dimension::provider::units_essentials::UnitsEssentialsV1Marker;
+
+use icu::experimental::dimension::provider::units_essentials::UnitsEssentialsV1;
+use icu_provider::prelude::*;
+use icu_provider::DataMarkerAttributes;
+use zerovec::ZeroMap;
+
+impl DataProvider<UnitsEssentialsV1Marker> for DatagenProvider {
+    fn load(&self, req: DataRequest) -> Result<DataResponse<UnitsEssentialsV1Marker>, DataError> {
+        self.check_req::<UnitsEssentialsV1Marker>(req)?;
+
+        // Get langid and the unit.
+        let langid = req.locale.get_langid();
+        let length = match req.marker_attributes.parse::<String>() {
+            Ok(aux_keys) => aux_keys,
+            Err(_) => return Err(DataError::custom("Failed to get aux keys")),
+        };
+
+        // Get units
+        let units_format_data: &cldr_serde::units::data::Resource =
+            self.cldr()?.units().read_and_parse(&langid, "units.json")?;
+        let units_format_data = &units_format_data.main.value.units;
+        let length_data = match length.as_str() {
+            "long" => &units_format_data.long,
+            "short" => &units_format_data.short,
+            "narrow" => &units_format_data.narrow,
+            _ => return Err(DataError::custom("Failed to get length data")),
+        };
+        let per = length_data
+            .get("per")
+            .and_then(|unit| unit.compound_unit_pattern.as_ref())
+            .ok_or_else(|| DataError::custom("Failed to get per"))?
+            .clone();
+
+        let times = length_data
+            .get("times")
+            .and_then(|unit| unit.compound_unit_pattern.as_ref())
+            .ok_or_else(|| DataError::custom("Failed to get times"))?
+            .clone();
+
+        // TODO: Fill prefixes (power, si prefixes, ... etc.) in the next PR.
+        let prefixes = ZeroMap::new();
+
+        let result = UnitsEssentialsV1 {
+            per: per.into(),
```

Comment:
issue: this is not using patterns

Reply:
yes, we added a todo in the `UnitsEssentialsV1` to use `DoublePlaceholderPattern`.